### PR TITLE
Bump Deno

### PR DESCRIPTION
**Pull request by lishaduck**
I'm not sure if there's a required amount of time before you bump Deno to ensure stability, but I didn't see anything, thus I went ahead and bumped it.

### CODE REVIEW

**Comment by lishaduck** [comment]
Ping, @TooTallNate :)
I think Deno 44 should be stable enough by now.

**Comment by TooTallNate** [comment]
I'm seeing a failure on the "dump" example endpoint. Looks like we'll need to start shipping the `deno.lock` file as part of the Lambda:

```
error: Failed writing lockfile.
Caused by:
Read-only file system (os error 30) (for '/var/task/deno.lock')
Child exited with error exit status 1INIT_REPORT Init Duration: 602.12 ms	Phase: init	Status: error	Error Type: Runtime.ExitError
```

**Comment by lishaduck** [comment]
Huh. That's weird. I don't have the "authorization required to deploy," yet it still deployed. ¯\_(ツ)_/¯

**Comment by lishaduck** [comment]
Ok, when you run deno, you need to add the `--no-lock` flag following _1.29_, but maybe they switched another default?

**Comment by lishaduck** [comment]
May I ask where those logs came from? Vercel? I'm not seeing how that's getting run.

**Comment by TooTallNate** [comment]
> Huh. That's weird. I don't have the "authorization required to deploy," yet it still deployed. ¯\_(ツ)_/¯

I had to manually approve it.

> May I ask where those logs came from?

Yes, they were from the Vercel Logs tab.

I think it's fine to have the lockfile, but we'll need to add it to the [list of `files`](https://github.com/vercel-community/deno/blob/main/src/deno-lambda.ts#L141-L149) so that the file is present in the Lambda at runtime.

**Comment by lishaduck** [comment]
> > Huh. That's weird. I don't have the "authorization required to deploy," yet it still deployed. ¯_(ツ)_/¯
> 
> I had to manually approve it.

Ah. That makes sense.

> > May I ask where those logs came from?
> 
> Yes, they were from the Vercel Logs tab.
> 
> I think it's fine to have the lockfile, but we'll need to add it to the [list of `files`](https://github.com/vercel-community/deno/blob/main/src/deno-lambda.ts#L141-L149) so that the file is present in the Lambda at runtime.

Ok, I can add it.

File: src/deno-lambda.ts [hunk 1]
```diff
@@ -1,7 +1,7 @@
 /**
  * The default version of Deno that will be downloaded at build-time.
  */
-const DEFAULT_DENO_VERSION = 'v1.42.1';
+const DEFAULT_DENO_VERSION = 'v1.44.4';
```
Comment (lishaduck) [inline src/deno-lambda.ts:4]: Bumps Deno

File: src/deno-lambda.ts [hunk 2]
```diff
@@ -146,6 +146,9 @@ export class DenoLambda extends Lambda {
 			'bin/deno': await FileFsRef.fromFsPath({
 				fsPath: join(runtimeDeno.dir, 'deno'),
 			}),
+			'deno.lock': await FileFsRef.fromFsPath({
+				fsPath: join(cwd, 'lock.json'),
+			}),
```
Comment (lishaduck) [inline src/deno-lambda.ts]: Fixes new behavior, I hope.

File: src/deno-lambda.ts [hunk 3]
```diff
@@ -146,6 +146,9 @@ export class DenoLambda extends Lambda {
 			'bin/deno': await FileFsRef.fromFsPath({
 				fsPath: join(runtimeDeno.dir, 'deno'),
 			}),
+			'deno.lock': await FileFsRef.fromFsPath({
+				fsPath: join(cwd, 'deno.lock'),
+			}),
```
Comment (lishaduck) [inline src/deno-lambda.ts:151]: Hopefully fixes the `dump/` issue.

File: package.json [hunk 4]
```diff
@@ -73,5 +73,6 @@
     "testMatch": [
       "<rootDir>/test/**/*.test.ts"
     ]
-  }
+  },
+  "packageManager": "pnpm@8.5.0"
```
Comment (lishaduck) [inline package.json:77]: Allow installing pnpm via corepack, which is how I install it, and is the recommended way if you don't like `curl | sh`.

File: .github/workflows/test.yml [hunk 5]
```diff
@@ -17,9 +17,7 @@ jobs:
     - uses: actions/checkout@v3
 
     - name: Install pnpm
-      uses: pnpm/action-setup@v2
-      with:
-        version: 8.5.0
+      uses: pnpm/action-setup@v4
```
Comment (lishaduck) [inline .github/workflows/test.yml:20]: This was a one-line change, so I bumped it while I was deleted the unnecessary version parameter (it'll read from `package.json`)

File: .prettierrc [hunk 6]
```diff
@@ -0,0 +1,3 @@
+{
+	"singleQuote": true
```
Comment (lishaduck) [inline .prettierrc:2]: So that Prettier didn't switch the file to double quotes.

File: src/deno-lambda.ts (hunk 3 above)
Comment (TooTallNate) [reply src/deno-lambda.ts:151]: Since `deno.lock` is not necessarily always present, pleased add a try/catch for this.

**Review by TooTallNate — CHANGES_REQUESTED** [review]

File: src/deno-lambda.ts (hunk 3 above)
Comment (lishaduck) [reply src/deno-lambda.ts]: Done! Anything else?

**Pull request merged**